element_array_ephys/ephys_no_curation.py

```diff
@@ -0,0 +1,863 @@
+        for ephys_pattern, ephys_acq_type in zip(['*.ap.meta', '*.oebin'],
+                                                 ['SpikeGLX', 'Open Ephys']):
+            ephys_meta_filepaths = list(sess_dir.rglob(ephys_pattern))
+            if ephys_meta_filepaths:
+                acq_software = ephys_acq_type
+                break
+        else:
+            raise FileNotFoundError(
+                f'Ephys recording data not found!'
+                f' Neither SpikeGLX nor Open Ephys recording files found')
+
+        if acq_software == 'SpikeGLX':
+            for meta_filepath in ephys_meta_filepaths:
+                spikeglx_meta = spikeglx.SpikeGLXMeta(meta_filepath)
+                if str(spikeglx_meta.probe_SN) == inserted_probe_serial_number:
+                    break
+            else:
+                raise FileNotFoundError(
+                    'No SpikeGLX data found for probe insertion: {}'.format(key))
+
+            if re.search('(1.0|2.0)', spikeglx_meta.probe_model):
+                probe_type = spikeglx_meta.probe_model
+                electrode_query = probe.ProbeType.Electrode & {'probe_type': probe_type}
+
+                probe_electrodes = {
+                    (shank, shank_col, shank_row): key
+                    for key, shank, shank_col, shank_row in zip(*electrode_query.fetch(
+                        'KEY', 'shank', 'shank_col', 'shank_row'))}
+
+                electrode_group_members = [
+                    probe_electrodes[(shank, shank_col, shank_row)]
+                    for shank, shank_col, shank_row, _ in spikeglx_meta.shankmap['data']]
+            else:
+                raise NotImplementedError(
+                    'Processing for neuropixels probe model'
+                    ' {} not yet implemented'.format(spikeglx_meta.probe_model))
+
+            self.insert1({**key,
+                          **generate_electrode_config(probe_type, electrode_group_members),
+                          'acq_software': acq_software,
+                          'sampling_rate': spikeglx_meta.meta['imSampRate'],
+                          'recording_datetime': spikeglx_meta.recording_time,
+                          'recording_duration': spikeglx_meta.recording_duration})
+
+            root_dir = find_root_directory(get_ephys_root_data_dir(), meta_filepath)
+            self.EphysFile.insert1({
+                **key,
+                'file_path': meta_filepath.relative_to(root_dir).as_posix()})
+        elif acq_software == 'Open Ephys':
+            dataset = openephys.OpenEphys(sess_dir)
+            for serial_number, probe_data in dataset.probes.items():
+                if str(serial_number) == inserted_probe_serial_number:
+                    break
+            else:
+                raise FileNotFoundError(
+                    'No Open Ephys data found for probe insertion: {}'.format(key))
+
+            if re.search('(1.0|2.0)', probe_data.probe_model):
```

Comment:
In regular expressions, `.` matches any one character.
```suggestion
            if re.search('(1|2)\.0', probe_data.probe_model):
```

Reply:
ah yes, forgot to escape that `.`